filament/backend/src/webgpu/WebGPUDriver.cpp

```diff
@@ -310,13 +310,11 @@ void WebGPUDriver::destroyRenderTarget(Handle<HwRenderTarget> rth) {
 }
 
 void WebGPUDriver::destroySwapChain(Handle<HwSwapChain> sch) {
-    mSwapChain = nullptr;
-    // TODO:  use webgpu handle allocator from
-    //        https://github.com/google/filament/pull/8566
-//    if (sch) {
-//        HwSwapChain* hwSwapChain = handleCast<HwSwapChain*>(sch);
-//        destruct(sch, hwSwapChain);
-//    }
+    if (sch) {
+        auto* hwSwapChain = handleCast<WGPUHwSwapChain>(sch);
+        destruct(sch, hwSwapChain);
+    }
```

Comment:
and then if `destruct` is changed then this could maybe just be 
```suggestion
     if (sch) {
        destructHandle<HwSwapChain>(sch);
    }
```